substrate/frame/salary/src/lib.rs

```diff
@@ -382,6 +394,54 @@ pub mod pallet {
 		}
 	}
 
+	#[pallet::tasks_experimental]
+	impl<T: Config<I>, I: 'static> Pallet<T, I> {
+		#[pallet::task_list(vec![].into_iter())]
+		#[pallet::task_condition(|| {
+			let now = frame_system::Pallet::<T>::block_number();
+			let cycle_period = Pallet::<T, I>::cycle_period();
+			let mut status = Status::<T, I>::get().unwrap();
+			status.cycle_start.saturating_accrue(cycle_period);
+			now >= status.cycle_start
+		})]
+		#[pallet::task_weight(T::DbWeight::get().reads(1))]
+		#[pallet::task_index(0)]
+		pub fn bump_offchain() -> DispatchResult {
+			let mut status = Status::<T, I>::get().ok_or(Error::<T, I>::NotStarted)?;
+			status.cycle_index.saturating_inc();
+			status.cycle_start = frame_system::Pallet::<T>::block_number();
+			status.budget = T::Budget::get();
+			status.total_registrations = Zero::zero();
+			status.total_unregistered_paid = Zero::zero();
+			Status::<T, I>::put(&status);
+
+			Pallet::deposit_event(Event::<T, I>::CycleStarted { index: status.cycle_index });
+			Ok(())
+		}
+	}
+
+	#[pallet::hooks]
+	impl<T: Config<I>, I: 'static> Hooks<BlockNumberFor<T>> for Pallet<T, I> {
+		#[cfg(feature = "experimental")]
+		fn offchain_worker(_block_number: BlockNumberFor<T>) {
+			// Create a valid task
+			let task = Task::<T, I>::BumpOffchain {};
+			let runtime_task = <T as Config<I>>::RuntimeTask::from(task);
+			let call = frame_system::Call::<T>::do_task { task: runtime_task.into() };
+
+			// Submit the task as an unsigned transaction
+			let xt = <T as CreateInherent<frame_system::Call<T>>>::create_inherent(call.into());
+			let res = SubmitTransaction::<T, frame_system::Call<T>>::submit_transaction(xt);
+			match res {
+				Ok(_) => log::info!(target: LOG_TARGET, "Submitted the task."),
+				Err(e) => log::error!(target: LOG_TARGET, "Error submitting task: {:?}", e),
+			}
```

Comment:
So every block we try to submit the task?

Is submitting a transaction so cheap? Maybe we should at least verify the task condition before trying to submit it.

maybe we can use the task list here.

Reply:
@gupnik what is the recommended way to submit tasks? every pallet do its own offchain worker that calls `Task::iter()`? or is there some system offchain worker responsible for this.

Reply:
I still don't know about this. I think system should have an offchain worker that calls `Task::iter()`, otherwise what is the point of the iter method.